csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -22,98 +22,91 @@ namespace Apache.Arrow
 {
     public partial class Schema
     {
-        public IReadOnlyDictionary<string, Field> Fields
-        {
-            get => _fieldsDictionary;
-        }
-
+        [Obsolete("Use `FieldsList` instead")]
```

Comment:
```suggestion
        [Obsolete("Use `FieldsLookup` instead")]
```

Reply:
Since `FieldsList` wasn't previously exposed, `Fields` was the only way of getting a list of all the fields (via `Fields.Values`). 

So for some use cases `FieldsList` would be preferred over `Fields`, but for others `FieldsLookup` would be preferred over `Fields`. I've updated the message to mention both, so the user can decide which would be best to use for their use case.